db-engine/src/main/java/com/splicemachine/db/iapi/types/SQLTimestamp.java

```diff
@@ -1584,6 +1586,49 @@ public void write(UnsafeRowWriter unsafeRowWriter, int ordinal) {
 				holder.cursor = 12;
 			}
 	    }
+
+	/**
+	 *
+	 * Write Element into Positioned Array
+	 *
+	 * @param unsafeArrayWriter
+	 * @param ordinal
+	 * @throws StandardException
+     */
+	@Override
+	public void writeArray(UnsafeArrayWriter unsafeArrayWriter, int ordinal) throws StandardException {
+		if (isNull())
+			unsafeArrayWriter.setNull(ordinal);
+		else {
+			byte[] data = new byte[12];
```

Comment:
No, I was curious about how the data is being written by the next statements like Patform.puInt(data,16,...) Isn't 16 the offset, which I thought, 16th byte of data-array it would start writing? pardon my ignorance, I am not too familiar with spark's Platform APIs.